src/dataframe/dataframe.jl

```diff
@@ -901,11 +922,7 @@ copies of column vectors in `df`.
 If `copycols=false`, return a new `DataFrame` sharing column vectors with `df`.
 """
 function Base.copy(df::DataFrame; copycols::Bool=true)
-    if copycols
-        df[:, :]
-    else
-        DataFrame(_columns(df), _names(df), copycols=false)
-    end
+    return DataFrame(copy(_columns(df)), copy(index(df)), copycols=copycols)
```

Comment:
Wait so multithreading isn't used to copy columns?

Reply:
It is used to copy columns - we just do not need to implement it here. `DataFrame` constructor uses threads. Note that `copy(_columns(df))` is a shallow copy only.

Reply:
though - what we could do is to sidestep the constructor at all (because it does some extra checks that are not needed in this case - e.g. here we know all columns have the same length, are 1-based, and are vectors). This potentially could give us some nanoseconds of savings - I was not sure if it is worth to do it though.

Reply:
Ah, sorry, it was late. Of course `copy(_columns(df))` doesn't "copy columns", it copies the vector of columns. :-)